dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/dxf2/events/aggregates/TrackedEntityInstanceAggregateTest.java

```diff
@@ -492,6 +490,15 @@ void testEventMapping()
         assertThat( event.getCompletedBy(), is( "[Unknown]" ) );
     }
 
+    private void assertUserProperties( Event event )
+    {
+        assertThat( event.getAssignedUserFirstName(), is( FIRST_NAME + TEST_USER ) );
```

Comment:
Since this test class uses hamcrest I get that you adhere to the style. I would do the same 😋 In this case what you wrote can and should be replaced with `assertEquals`. In the backend team we discussed that we will not use hamcrest just for the sake of its syntactic sugar but stick with JUnit 5 assertions (where possible).

See https://github.com/dhis2/wow-backend/blob/master/guides/testing.md#assertions

https://github.com/dhis2/dhis2-core/blob/master/dhis-2/dhis-support/dhis-support-test/src/main/java/org/hisp/dhis/utils/Assertions.java is where we try to build up useful assertions mainly based on JUnit assertions. If JUnit does not provide an assertion and another library has one that is very useful we would create a wrapper in the Assertions class.

Reply:
good to know, thanks!